library/ssl_tls13_generic.c

```diff
@@ -922,6 +922,14 @@ int mbedtls_ssl_parse_signature_algorithms_ext( mbedtls_ssl_context *ssl,
                                         size_t len )
 {
     size_t sig_alg_list_size; /* size of receive signature algorithms list */
+    sig_alg_list_size = ( ( buf[0] << 8 ) | ( buf[1] ) );
```

Comment:
Upstream Mbed TLS has recently moved to C99, but nonetheless most if not all of the code puts variable declarations at the beginning of scopes, which isn't the case with this change if `MBEDTLS_SSL_SRV_C` is defined. Could you adapt the code in a way that keeps declarations at the top of the scope?